gate-plugins/src/main/kotlin/com/netflix/spinnaker/gate/plugins/web/publish/PluginPublishController.kt

```diff
@@ -26,33 +27,39 @@ import okhttp3.MediaType
 import okhttp3.MultipartBody
 import okhttp3.OkHttpClient
 import okhttp3.Request
-import okhttp3.RequestBody
+import okhttp3.RequestBody as OkHttpRequestBody
+import org.springframework.http.MediaType.MULTIPART_FORM_DATA_VALUE
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RequestPart
 import org.springframework.web.bind.annotation.RestController
 import org.springframework.web.multipart.MultipartFile
 
 @RestController
-@RequestMapping("/plugins/upload")
-class PluginBinaryController(
+@RequestMapping("/plugin/publish")
```

Comment:
I decided on the singular `/plugin` instead of `/plugins`.  `/plugins` is currently used here: https://github.com/spinnaker/kork/blob/master/kork-plugins/src/main/java/com/netflix/spinnaker/kork/plugins/spring/actuator/SpinnakerPluginEndpoint.java

As you can imagine that makes things confusing.  I also think the singular `/plugin` generally makes more sense.

Reply:
We should change the actuator endpoint — thats internal. I think I’d rather have a clear public api?

Reply:
Yeah.  Should I go back to `/plugins` here and then change the actuator endpoint to something else?  Or leave `/plugin` here but still change the actuator endpoint?

Reply:
I'd say change the actuator endpoint and go back to `/plugins`.

Reply:
Sounds good.

Reply:
https://github.com/spinnaker/kork/pull/656